op-node/rollup/derive/batch_queue.go

```diff
@@ -60,19 +67,58 @@ func (bq *BatchQueue) Progress() Progress {
 }
 
 func (bq *BatchQueue) Step(ctx context.Context, outer Progress) error {
-	if changed, err := bq.progress.Update(outer); err != nil {
-		return err
-	} else if changed {
-		if !bq.progress.Closed { // init inputs if we moved to a new open origin
+
+	originBehind := bq.progress.Origin.Number < bq.next.SafeL2Head().L1Origin.Number
```

Comment:
Assuming the pipeline reset logic is walking back the L1 header chain starting from the L2 safe head and checking all L1 origins parent hashes this should be safe, but it wouldn't hurt to do a consistency check to make sure the safe head data is consistent with the L1 origin when we need it to, to avoid weird reorg edge cases (I hope to capture these soon in the action testing). Outside of the scope of this PR if we focus on channel bank + channel reader here, but worth looking at.